vue/src/lib/button/button.vue

```diff
@@ -0,0 +1,222 @@
+    /** @prop Sets active css styling | false */
+    active: Boolean,
+    /** @prop Text to display for blindness accessibility features | '' */
+    ariaLabel: {
+      type: String,
+      default: ''
+    },
+    /** @prop ID to reference for blindness accessibility feature | '' */
+    ariaLabelledBy: {
+      type: String,
+      default: ''
+    },
+    /** @prop Sets circle css styling | false */
+    circle: Boolean,
+    /** @prop Optional css class string | '' */
+    className: {
+      type: String,
+      default: ''
+    },
+    /** @prop Sets optional Button color | '' */
+    color: {
+      type: String,
+      default: ''
+    },
+    /** @prop Sets containerLarge css styling | false */
+    containerLarge: Boolean,
+    /** @prop Sets the attribute disabled to Button | false */
+    disabled: Boolean,
+    /** @prop Sets expand css styling to widen the Button | false */
+    expand: Boolean,
+    /** @prop Href prop changes element to anchor element | '' */
+    href: {
+      type: String,
+      default: ''
+    },
+    /** @prop This index is used to control focus of Button within a ButtonGroup | null */
+    index: Number,
+    /** @prop Text to display inside the button | '' */
+    label: {
+      type: String,
+      default: ''
+    },
+    /** @prop Activates the loading animation and sets the button to disabled | false */
+    loading: Boolean,
+    /** @prop Optional prop to remove Button's default style | false */
+    removeStyle: Boolean,
+    /** @prop Optional string or number size prop | 36 */
+    size: {
+      type: [String, Number],
+      default: 36
+    },
+    /** @prop Additional css styling applied to the button | {} */
+    cssStyle: {
+      type: Object,
+      default: function () {
+        return {}
+      }
+    },
+    /** @prop Optional tag to define type of element | 'button' */
+    tag: {
+      type: String,
+      default: 'button',
+      validator: function (value) {
+        return ['button', 'input', 'a'].indexOf(value) !== -1
+      }
+    },
+    /** @prop Optional html type | 'button' */
+    type: {
+      type: String,
+      default: 'button',
+      validator: function (value) {
+        return ['button', 'reset', 'submit'].indexOf(value) !== -1
+      }
+    },
+  },
+
+  mounted() {
+    (!this.ariaLabel && !this.ariaLabelledBy)
+      &&
+      console.warn('[@momentum-ui/vue] Button: Accessibility could be improved with ariaLabel');
+
+    this.focusOnLoad
+      && this.focusIndex === this.index
+      && this.$refs.button.focus();
+  },
+
+  updated() {
+    typeof this.index === 'number'
+      && this.index === this.focusIndex
+      && this.$refs.button.focus();
+  },
+
+  methods: {
+    handleClick(e) {
+      this.$emit('click', e);
+      this.click && this.click(e, this.index);
+    },
+
+    handleKeyDown(e) {
+      if (e.which === 32 || e.which === 13 ||
+          e.charCode === 32 || e.charCode === 13) {
+        this.click && this.click(e, this.index);
```

Comment:
event.key || event.which || event.keyCode i believe is what we are trying to keep using for keydown events; charCode is deprecated. @bfbiggs

Reply:
added keyCode, but still need charCode for Jest unit test.